model/device.go

```diff
@@ -78,6 +136,37 @@ func (da DeviceAttribute) Validate() error {
 	)
 }
 
+func (da DeviceAttribute) Equal(e DeviceAttribute) bool {
+	if (da.Value == nil || e.Value == nil) && !(da.Value == nil && e.Value == nil) {
+		return false
+	}
+	ok1 := reflect.TypeOf(da.Value).Kind() == reflect.Slice
+	ok2 := reflect.TypeOf(e.Value).Kind() == reflect.Slice
+	if ok1 || ok2 {
+		// there is no portable way to compare slices of unknown type
+		return false
+	} else {
+		if da.Value != e.Value {
+			return false
+		}
+	}
+
+	if da.Scope == e.Scope {
+		if da.Name == e.Name {
+			if da.Description == nil && e.Description == nil {
+				return true
+			}
+			if da.Description == nil || e.Description == nil {
+				return false
+			}
+			if *da.Description == *e.Description {
+				return true
+			}
```

Comment:
The above check asserts that the attributes are not equal if both descriptions are empty (nil).
```suggestion
			if da.Description != nil {
				if e.Description == nil {
					return false
				} else if *da.Description != *e.Description {
					return false
				}
			}
```

Reply:
>The above check asserts that the attributes are not equal if both descriptions are empty (nil).

if the descriptions are nil the assumption is they are equal (see below for my reasoning)

```go
				if da.Description == nil && e.Description == nil {
					// at this point Value, Scope, and Name are equal, and both Descriptions are nil
					// shall I assume otherwise?
					return true
				}
				if da.Description == nil || e.Description == nil {
					// if either is nil while the other is not they are different
					return false
				}
				if *da.Description == *e.Description {
					// both not nil, we can compare and if equal the attributes are equal
					return true
				}
```